[NNC] Add Softplus operator

anijain2305 opened the pull request:
Summary: Adding softplus operator lowering for NNC. Enabling element wise fusion as well.

Test Plan: Added a test in test_jit_fuser.py

Differential Revision: D30736449

bertmaher commented on test/test_jit_fuser_te.py:
```diff
@@ -1255,6 +1255,7 @@ def apply(fn):
             F.hardtanh,
             F.hardsigmoid,
             F.hardswish,
+            F.softplus,
```
Based on test failures, it looks like this should also be added to `works_list` lower in this file

anijain2305 replied:
Added later.

bertmaher reviewed (changes requested):
Other than the failing test this looks good, please update the test and I'll happily approve!

bertmaher commented:
It would be cool to run these benchmarks and see if the fused `mish` is now on par with the native implementation: https://benjaminwarner.dev/2021/07/19/benchmarking-pytorch-native-mish

anijain2305 committed:
[NNC] Add Softplus operator (#64589)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/64589

Adding softplus operator lowering for NNC. Enabling element wise fusion as well.

Test Plan: Added a test in test_jit_fuser.py

Differential Revision: D30736449

fbshipit-source-id: d809757fb002822dc145acb3bbf2beba8efd876e

bertmaher reviewed (approved):
Nice work, thanks!
